user_modules/IsReady_Template/testCasesForUI_IsReady.lua

```diff
@@ -0,0 +1,272 @@
+local testCasesForUI_IsReady = {}
+
+require('atf.util')
+local events = require("events")
+
+--[[@InitHMI_onReady_without_UI_IsReady_available_false: replace original InitHMIOnReady from connecttest
+--! without expect UI.IsReady
+--! @parameters: exp_occur - times to wait UI.<<RPC>>
+--! in case UI.IsReady(available = false), exp_occur = 0, else exp_occur = 1
+--]]
+function testCasesForUI_IsReady.InitHMI_onReady_without_UI_IsReady(self, exp_occur)
+  if(exp_occur == nil) then exp_occur = 0 end
+
+  local function ExpectRequest(name, mandatory, params)
+    local event = events.Event()
+    event.level = 2
+    event.matches = function(_, data) return data.method == name end
+    return
+      EXPECT_HMIEVENT(event, name)
+      :Times(mandatory and 1 or AnyNumber())
+      :Do(function(_, data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", params)
+      end)
+  end
+
+  ExpectRequest("BasicCommunication.MixingAudioSupported",
+    true,
+    { attenuatedSupported = true })
+  ExpectRequest("BasicCommunication.GetSystemInfo", false,
+    {
+      ccpu_version = "ccpu_version",
+      language = "EN-US",
+      wersCountryCode = "wersCountryCode"
+    })
+  ExpectRequest("UI.GetLanguage", true, { language = "EN-US" })
+  :Times(exp_occur)
+  :Timeout(20000)
```

Comment:
@istoimenova, Please note currently if Times(0) is used Timeout(x) function won't work. It seems this is an ATF issue and needs to be fixed.
As workaround I propose to use commonTestCases:DelayedExp() function and define timeout for the whole InitHMI_onReady_without_UI_IsReady() function